presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -142,8 +141,7 @@ public static ParquetPageSource createParquetPageSource(
             boolean predicatePushdownEnabled,
             TupleDomain<HiveColumnHandle> effectivePredicate)
     {
-        ParquetDataSource dataSource = buildHdfsParquetDataSource(path, configuration, start, length);
-        try {
+        try (ParquetDataSource dataSource = buildHdfsParquetDataSource(path, configuration, start, length)) {
```

Comment:
This will cause the data source to be closed when exiting this block.  Instead, you should take a look at how OrcDataSource is closed.